R/batchmark.R

```diff
@@ -0,0 +1,182 @@
+#' @param reg [\code{\link[batchtools]{Registry}}]\cr
+#'   Registry, created by \code{\link[batchtools]{makeExperimentRegistry}}. If not explicitly passed, 
+#'   uses the last created registry.
+#' @return [\code{data.table}]. Generated job ids are stored in the column \dQuote{job.id}.
+#' @export
+#' @family benchmark
+batchmark = function(learners, tasks, data.ids, resamplings, measures, models = TRUE, reg = batchtools::getDefaultRegistry()) {
+  
+  requirePackages("batchtools", why = "batchmark", default.method = "load")
+  
+  learners = ensureBenchmarkLearners(learners)
+  learner.ids = extractSubList(learners, "id")
+  names(learners) = learner.ids
+  
+  
+
+  if (!missing(tasks)) {
+    tasks = ensureBenchmarkTasks(tasks)
+    task.ids = extractSubList(tasks, c("task.desc", "id"))
+    names(tasks) = task.ids
+  }
+  if (!missing(data.ids)) {
+    requirePackages("OpenML", why = "OpenML data ids given", default.method = "load")
+    OpenML::populateOMLCache(data.ids = data.ids)
+    did.tasks = as.list(data.ids)
+    names(did.tasks) = data.ids
+    tasks = if (missing(tasks))
+      did.tasks
+    else
+      c(tasks, did.tasks)
+    task.ids = names(tasks)
+  } 
+  if (missing(tasks))
+    stop("At least one task or OpenML data ID has to be given")
+
+  
+  resamplings = ensureBenchmarkResamplings(resamplings, tasks)
+  names(resamplings) = task.ids
+  
+  measures = ensureBenchmarkMeasures(measures, tasks)
+  
+  assertFlag(models)
+
+  reg$packages = union(reg$packages, "mlr")
+  
+  #set name of learner list
+  lrn.ids = vapply(learners, getLearnerId, FUN.VALUE = character(1))
+  names(learners) = lrn.ids
+
+  # generate problems
+  pdes = Map(function(id, task, rdesc, seed) {
+    batchtools::addProblem(id, data = list(rdesc = rdesc, task = task, measures = measures, learners = learners), fun = resample.fun, seed = seed, reg = reg)
+    data.table(i = seq_len(rdesc$iters))
+  }, id = task.ids, task = tasks, rdesc = resamplings, seed = reg$seed + seq_along(tasks))
+
+  # generate algos
+  ades = Map(function(id, learner) {
+    apply.fun = getAlgoFun(learner, measures, models)
+    batchtools::addAlgorithm(id, apply.fun, reg = reg)
+    data.table()
+  }, id = learner.ids, learner = learners)
+
+  # add experiments
+  batchtools::addExperiments(reg = reg, prob.designs = pdes, algo.designs = ades, repls = 1L)
+}
+
+resample.fun = function(job, data, i) {
+  if (is.numeric(data$task))
+    data$task = OpenML::convertOMLDataSetToMlr(getOMLDataSet(data$task))
+  rin = makeResampleInstance(desc = data$rdesc, task = data$task)
+  list(train = rin$train.inds[[i]], test = rin$test.inds[[i]], weights = rin$weights[[i]])
+}
+
+getAlgoFun = function(lrn, measures, models) {
+  force(lrn)
+  force(measures)
+  force(models)
+  function(job, data, instance) {
+
+    extract.this = getExtractor(lrn)
+    if (is.numeric(data$task))
+      data$task = OpenML::convertOMLDataSetToMlr(getOMLDataSet(data$task))
```

Comment:
Yes, it should be in the OML cache anyways so we do not want/need to save it for every job.